tests/views/test_admin_acls.py

```diff
@@ -0,0 +1,171 @@
+            res.context['existing_acls'][0].email_address, 'l@mib.gov')
+
+    def test_edit_access_level(self):
+        """Tests editing the access level for a user."""
+        self.data_generator.admin_permission(
+            email_address='j@mib.gov',
+            expiration_date=datetime.datetime(2019, 4, 25),
+            access_level=admin_acls_model.AdminPermission.AccessLevel.MODERATOR)
+        get_doc = self.to_doc(self.client.get(
+            '/haiti/admin/acls/', secure=True))
+        xsrf_token = get_doc.cssselect('input[name="xsrf_token"]')[1].get(
+            'value')
+        post_resp = self.client.post('/haiti/admin/acls', {
+            'xsrf_token': xsrf_token,
+            'email_address': 'j@mib.gov',
+            'expiration_date': '2019-04-25',
+            'level': 'administrator',
+        }, secure=True)
+        acls = admin_acls_model.AdminPermission.all()
+        self.assertEqual(acls.count(), 1)
+        acl = acls[0]
+        self.assertEqual(acl.email_address, 'j@mib.gov')
+        self.assertEqual(acl.expiration_date, datetime.datetime(2019, 4, 25))
+        self.assertEqual(
+            acl.access_level,
+            admin_acls_model.AdminPermission.AccessLevel.ADMINISTRATOR)
+
+    def test_edit_expiration_date(self):
+        """Tests editing the expiration date for a permission."""
+        self.data_generator.admin_permission(
+            email_address='j@mib.gov',
+            expiration_date=datetime.datetime(2019, 4, 25),
+            access_level=admin_acls_model.AdminPermission.AccessLevel.MODERATOR)
+        get_doc = self.to_doc(self.client.get(
+            '/haiti/admin/acls/', secure=True))
+        xsrf_token = get_doc.cssselect('input[name="xsrf_token"]')[1].get(
+            'value')
+        post_resp = self.client.post('/haiti/admin/acls', {
+            'xsrf_token': xsrf_token,
+            'email_address': 'j@mib.gov',
+            'expiration_date': '2019-05-25',
+            'level': 'moderator',
+        }, secure=True)
+        acls = admin_acls_model.AdminPermission.all()
+        self.assertEqual(acls.count(), 1)
+        acl = acls[0]
+        self.assertEqual(acl.email_address, 'j@mib.gov')
+        self.assertEqual(acl.expiration_date, datetime.datetime(2019, 5, 25))
+        self.assertEqual(
+            acl.access_level,
+            admin_acls_model.AdminPermission.AccessLevel.MODERATOR)
+
+    def test_revoke_access(self):
+        """Tests revoking admin access."""
+        self.data_generator.admin_permission(email_address='j@mib.gov')
+        get_doc = self.to_doc(self.client.get(
+            '/haiti/admin/acls/', secure=True))
+        xsrf_token = get_doc.cssselect('input[name="xsrf_token"]')[1].get(
+            'value')
+        post_resp = self.client.post('/haiti/admin/acls', {
+            'xsrf_token': xsrf_token,
+            'email_address': 'j@mib.gov',
+            'expiration_date': '2019-04-25',
+            'level': 'moderator',
```

Comment:
it looks like 'expiration_date' and 'level' are not checked by the server during the deletion - should they be removed from the unit test?

Reply:
The server always reads all of them at the beginning of the `post` function, so I need them to be present.